Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -1811,6 +1813,50 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
 }
 
 
+int CLuaPedDefs::SetPedStat ( lua_State* luaVM )
+{
+    // Verify the argument
+    CClientPed* pPed = NULL;
+    unsigned short usStat = 0;
+    float fValue = 0;
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadUserData ( pPed );
+    argStream.ReadNumber ( usStat );
+    argStream.ReadNumber ( fValue );
+
+    if ( !argStream.HasErrors ( ) )
+    {
+        // Only allow on local peds
+        if ( !pPed->IsLocalEntity ( ) )
+            argStream.SetCustomError ( "This client side function will only work with client created peds." );
+        
+        // Check the stat and value
+        if ( !argStream.HasErrors ( ) )
+            if ( usStat > NUM_PLAYER_STATS || usStat < 0 || fValue < 0.0f || fValue > 1000.0f )
+                argStream.SetCustomError ( "Stat must be 0 to 343 and value must be 0 to 1000." );
+               
+        // Dont let them set visual stats if they don't have the CJ model
+        if ( !argStream.HasErrors ( ) )
+            if ( ( usStat == 21 /* FAT */ || usStat == 23 /* BODY_MUSCLE */ ) && pPed->GetModel ( ) != 0 )
+                argStream.SetCustomError( "Fat and muscle stat can only be set on CJ skin." );
+
+        if ( !argStream.HasErrors ( ) )
+        {
+            pPed->SetStat ( usStat, fValue );
```

Comment:
This should also change the stats for child peds.

See [the server side implementation](https://github.com/multitheftauto/mtasa-blue/blob/master/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L3838)

You may need to create a similar method with [these lines](https://github.com/multitheftauto/mtasa-blue/blob/master/Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L3860-L3878).

You will need to add the new method [around here](https://github.com/multitheftauto/mtasa-blue/blob/master/Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp#L2170-L2185) (you can refer to the linked method as a simple example - you need not create an extra method checking for validity, but please do check for the CJ skin). Don't forget to update the header file too.

Ideally both the client/server side functions should behave exactly the same (with the only difference being that the client side function only works on client created peds).